lib/presentation/widget/biometric_authentication/biometric_authentication_setting_viewmodel.dart

```diff
@@ -0,0 +1,139 @@
+            .getOrElse(() => BiometricState.disabled);
+
+        response[1].fold((failure) {
+          store.dispatch(SetBiometricAuthenticationAction(biometricState, []));
+        }, (success) {
+          store.dispatch(SetBiometricAuthenticationAction(
+              biometricState,
+              success is GetAvailableBiometricViewState ? success.biometricKinds : []));
+        });
+      });
+    });
+  }
+
+  void backToAccountDetail() {
+    _appNavigation.popBack();
+    if (store.state.biometricAuthenticationSettingState.authenticationBiometricState == AuthenticationBiometricState.unEnrolled) {
+      store.dispatch(SetAuthenticationBiometricStateAction(AuthenticationBiometricState.unAuthenticated));
+    }
+  }
+
+  void checkBiometricState(BuildContext context) {
+    final authenticationBiometricState = store.state.biometricAuthenticationSettingState.authenticationBiometricState;
+    if (authenticationBiometricState == AuthenticationBiometricState.unEnrolled) {
+      _authenticationBiometric(context, BiometricState.disabled);
+    }
+  }
+
+  void toggleBiometricState(BuildContext context) {
+    final biometricState = store.state.biometricAuthenticationSettingState.biometricState;
+    _authenticationBiometric(context, biometricState);
+  }
+
+  void _authenticationBiometric(BuildContext context, BiometricState biometricState) {
+    store.dispatch((Store<AppState> store) async {
+      final localizedReason = AppLocalizations.of(context).biometric_authentication_localized_reason(
+          store.state.biometricAuthenticationSettingState.biometricKindList.getBiometricKind(context));
+      await _authenticationBiometricInteractor.execute(localizedReason)
+          .then((result) => result.fold(
+            (failure) => store.dispatch(SetAuthenticationBiometricStateAction(AuthenticationBiometricState.unAuthenticated)),
+            (success) {
+              if (success is AuthenticationBiometricViewState) {
+                if (success.authenticationState == AuthenticationBiometricState.authenticated) {
+                  final newBiometricState = biometricState == BiometricState.disabled ? BiometricState.enabled : BiometricState.disabled;
+
+                  newBiometricState == BiometricState.enabled
+                      ? _enableBiometricAuthentication(newBiometricState)
+                      : _disableBiometricAuthentication();
+
+                  store.dispatch(SetBiometricStateAction(newBiometricState));
+                }
+                store.dispatch(SetAuthenticationBiometricStateAction(success.authenticationState));
+              } else {
+                store.dispatch(SetAuthenticationBiometricStateAction(AuthenticationBiometricState.unAuthenticated));
+              }
+          }));
+    });
+  }
+
+  void _enableBiometricAuthentication(BiometricState biometricState) {
+    store.dispatch((Store<AppState> store) async {
+      await _enableBiometricInteractor.execute(biometricState);
```

Comment:
I don't really understand what you are trying to do there can you explain please ? 

You're dispatching a `Right<Failure, Success>(EnableBiometricViewState());`, you don't have to use an action ?

Reply:
Here I just save the biometric on and off state. The action I want to take after doing this is to display a notification to the user. But at the moment I don't see my design having that. So I create such an action temporarily. Maybe we will use it later.